### PR TITLE
Set up cookie refreshing skeleton

### PR DESCRIPTION
**Pull request by osdnk**
## Motivation
We need to have a mechanism to manage sessions refreshing. Namely, we need a mechanism to set up a private key and then make a signature for the challenge sent from the backend side.

## Changes
Again, as in #6, I decided to make a persisting part of the wallet itself to avoid deep integration with a Kolektivo codebase. That solution allows us to iterate fast and make the storing in an opinionated way. However, our decisions are limited to `CapsuleReactNativeWallet`, which means that the wallet developers can change our decisions and provide us with their own data management solution.

Again, I decided to make it a fake solution ot show how it can work and defer the actual implementation (which is tedious and includes native, platform-specific code) to a later stage. 

I decided to make a cookie parsing semi-manual to avoid integrating additional libraries. Also, while verifying the cookie validity, I added a 30s threshold. 

Yet, we cannot really test it properly cause we do not create a new wallet account (fixed by #8. After that, we need to call the function `refreshBiometricsIfNeeded` before every integration with the capsule signer.  

Also, we will need to remove the hardcoded userID after merging with #8.

### CODE REVIEW

File: src/capsule/CapsuleWallet.ts [hunk 1]
```diff
@@ -6,10 +6,43 @@ import * as ethUtil from 'ethereumjs-util'
...
 import Logger from 'src/utils/Logger'
+import { ChallengeStorageDefault } from './ChallengeStorage'
+import userManagementClient from './UserManagementClient'
 
 const TAG = 'geth/CapsuleWallet'
 
 export class CapsuleWallet extends RemoteWallet<CapsuleSigner> implements UnlockableWallet {
+  // TODO remove me
+  private userId = 'c67b0766-f339-4d86-9c82-fe2410b28460'
+  private biometricStorage = new ChallengeStorageDefault(this.userId)
```
Comment (osdnk) [inline src/capsule/CapsuleWallet.ts:17]: I expect we should move it to `CapsuleReactNativeWallet` implementation after merging #6

File: src/capsule/CapsuleWallet.ts [hunk 2]
```diff
@@ -6,10 +6,43 @@ import * as ethUtil from 'ethereumjs-util'
...
 import { CapsuleSigner } from 'src/capsule/CapsuleSigner'
 import Logger from 'src/utils/Logger'
+import { ChallengeStorageDefault } from './ChallengeStorage'
+import userManagementClient from './UserManagementClient'
 
 const TAG = 'geth/CapsuleWallet'
 
 export class CapsuleWallet extends RemoteWallet<CapsuleSigner> implements UnlockableWallet {
+  // TODO remove me
+  private userId = 'c67b0766-f339-4d86-9c82-fe2410b28460'
```
Comment (osdnk) [inline src/capsule/CapsuleWallet.ts:16]: Remove after merging #8

File: src/capsule/CapsuleWallet.ts [hunk 3]
```diff
@@ -6,10 +6,43 @@ import * as ethUtil from 'ethereumjs-util'
...
+      publicKey: await this.biometricStorage.getPublicKey(),
+    })
+  }
+
+  private cookie: string | undefined
+
+  public async refreshBiometricsIfNeeded() {
+    if (typeof this.cookie === 'string') {
+      // this is how cookie is represented. We do parsing "manually" to avoid employing additional libs
+      const expDate = this.cookie.split(';')[2].split('=')[1]
```
Comment (nityas) [inline src/capsule/CapsuleWallet.ts]: for some clarity could you add a comment with the expected format of the cookie?
Seems worth adding some validation checks in here

**Review by nityas — APPROVED** [review]
looks good to me- just a question on the cookie + minor request on parsing

**Pull request merged**